deployment/helm/templates/deployment.yaml

```diff
@@ -39,32 +43,36 @@ spec:
             httpGet:
               path: /
               port: http
-          {{- if or .Values.config.items .Values.config.main }}      
+          {{- if or .Values.config.items .Values.config.main }}
           volumeMounts:
           {{- end }}
-          {{- if .Values.config.main }}      
+          {{- if .Values.config.main }}
           - mountPath: /app/config/main.yaml
             subPath: main.yaml
             name: casavue-cfg-main
           {{- end }}
+          {{- if .Values.config.items }}
```

Comment:
This was unconditioned with purpose. When CasaVue starts, it looks for `main.yaml` and `items.yaml` in `./config/` directory. If either of the files is not found, it's is created with default content. This is done, so even when configuration is missing (e.g. first run), user is presented with meaningful view.

In case of `items.yaml` it means some default static entries, which in case of Kubernetes hosting environment is rather unwanted, because there are usually already some Ingresses on cluster (thus still giving meaningful view on first run) and keeping config to a minimum (not forcing user to create a ConfigMap with empty entries array).

So Helm Chart was designed to *always* mount static files, so after first install, hopefully user doesn't have to do any extra configuration.

Having that in mind, I do realize it could be counterintuitive. So what's your opinion, should there be demo static entries on default Helm Chart deployment? If so, we can leave this condition.

Reply:
I see, the problem (and why I put that in) was this doesn't work with a default install-

```
spirilis@apollo:~/k8s/casavue$ helm -n casavue upgrade -i casavue ./casavue/deployment/helm -f values.yaml
Error: UPGRADE FAILED: YAML parse error on casavue/templates/deployment.yaml: error converting YAML to JSON: yaml: line 51: did not find expected key
```

It might be better to implement a "fail" clause.  Let me see about that.

Reply:
Ok nevermind I finally figured out why that was happening.  It's ok to leave out any if statements for items b/c you populate it with an empty set in config_items.yaml if it's not there.  The problem is gating the presence of volumeMounts: and volumes: as requiring one of the config options to be there.  I took those if statements out and it makes more sense now (and works).

Btw, even with the empty set of items, I still get the default searching/social items anyway.  Might be a bug in the code?

Reply:
Sorry to hear that you had some unexpected behavior. The idea was for Helm Chart to create `config/items.yaml` with empty `items` array as a default, aiming for overriding  "demo" items.

I'm definitely going to look into this. Because it didn't come up on my tests, I'll appreciate all info that could help me replicate the problem - e.g. did you follow https://casavue.app/deployment/quick_start/, did you provide custom `values.yaml` file while deploying?

BTW when it comes to merging PR it self, I think we need to rollback on the `APP_VERSION.txt` bump, since it's equivalent to releasing a new container (which would be binary equal to existing version). But to be clear! - it's another signal for me, about lack of info about development and commiting in docs. I really didn't expect some community commits so fast (or ever :D) - a really pleasurable surprise, thanks, and sorry at the same time for having to face this shortcomings.

Reply:
> Btw, even with the empty set of items, I still get the default searching/social items anyway. Might be a bug in the code?

Hey, I caught a bug in WebUI that was causing unintentional caching of API requests. This may have affected your issues with static demo entries continuing to appear after config changes. I have released a new version of the app and chart: https://github.com/czoczo/casavue/pull/3 It should be better.